core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -240,15 +245,25 @@ public void finished(Description description) {
             ambassadorContainers.forEach((String address, AmbassadorContainer container) -> container.stop());
 
             // Kill the services using docker-compose
-            getDockerCompose("down -v")
-                    .start();
+            try {
+                getDockerCompose("down -v").invoke();
 
-            // remove the networks before removing the containers
-            ResourceReaper.instance().removeNetworks(identifier);
+                // If we reach here then docker-compose down has cleared networks and containers;
+                //  we can unregister from ResourceReaper
+                spawnedContainerIds.forEach(id -> ResourceReaper.instance().unregisterContainer(id));
```

Comment:
Not quite sure I follow this:
> Since we list networks in Docker Compose's implementation, maybe we can use the IDs there as well?

But I'd interpret this as 'let's only register network IDs, not names'. Is that right? I'll change and make sure we use IDs throughout, then I think we should be OK.